kotlin/openID4VP/src/main/java/io/mosip/openID4VP/authorizationRequest/AuthorizationRequest.kt

```diff
@@ -72,24 +79,111 @@ data class AuthorizationRequest(
             }
         }
 
-        private fun extractQueryParams(query: String): Map<String, String> {
+        private fun extractQueryParams(query: String): MutableMap<String, String> {
             try {
-                return query.split("&").map { it.split("=") }.associateBy({ it[0] }, {
+                return query.split("&").map { it.split("=") }
+                    .associateByTo(mutableMapOf(), { it[0] }, {
                     if (it.size > 1) URLDecoder.decode(
                         it[1], StandardCharsets.UTF_8.toString()
                     ) else ""
                 })
             } catch (exception: Exception) {
-                throw AuthorizationRequestExceptions.InvalidQueryParams("Exception occurred when extracting the query params from Authorization Request : ${exception.message}")
+                throw Logger.handleException(
+                    exceptionType = "InvalidQueryParams",
+                    message = "Exception occurred when extracting the query params from Authorization Request : ${exception.message}",
+                    className = className
+                )
             }
         }
 
+        private fun fetchPresentationDefinition(params: Map<String, String>): String {
+            val hasPresentationDefinition = params.containsKey("presentation_definition")
+            val hasPresentationDefinitionUri = params.containsKey("presentation_definition_uri")
+            var presentationDefinition = ""
+
+            when {
+                hasPresentationDefinition && hasPresentationDefinitionUri -> {
+                    throw Logger.handleException(
+                        exceptionType = "InvalidQueryParams",
+                        message = "Either presentation_definition or presentation_definition_uri request param can be provided but not both",
+                        className = className
+                    )
+                }
+
+                hasPresentationDefinition -> {
+                    val value = params["presentation_definition"] ?: throw Logger.handleException(
+                        exceptionType = "MissingInput",
+                        fieldPath = listOf("presentation_definition"),
+                        className = className
+                    )
+                    require(isNeitherNullNorEmpty(value)) {
+                        throw Logger.handleException(
+                            exceptionType = "InvalidInput",
+                            fieldPath = listOf("presentation_definition"),
+                            className = className
+                        )
+                    }
```

Comment:
All these scenarios are for Invalid input
param["presentation_definition"] = null 
param["presentation_definition"] = "null"
param["presentation_definition"] = ""

Reply:
Handled it and in first scenario library will take the null as String which is "null" because we are sending it as part of the request in "key1=value1&key2=value2" format and second scenario is invalid because when sending the presentation_definition as part of the request by default it's value will be considered as string so if we again pass the value wrapped in double quotes as presentation_definition="null" then library will considered "null" as the actual value instead of taking it as null. So for this scenario it will go ahead with the other validations.